packages/x-charts/src/hooks/useTicks.ts

```diff
@@ -74,12 +79,26 @@ function getTickPosition(
   );
 }
 
+/**
+ * Returns a new domain where each tick is at least {@link tickSpacing}px from the next one.
+ * Assumes tick spacing is greater than 0.
+ * @param domain Domain of the scale.
+ * @param range Range of the scale.
+ * @param tickSpacing Spacing in pixels.
+ */
+export function applyTickSpacing<T>(domain: T[], range: [number, number], tickSpacing: number) {
+  const rangeSpan = Math.abs(range[1] - range[0]);
+
+  const every = Math.ceil(domain.length / (rangeSpan / tickSpacing));
+  return domain.filter((_, index) => index % every === 0);
+}
```

Comment:
To handle the case tick spacing = 0

```suggestion
  const every = Math.ceil(domain.length * tickSpacing / rangeSpan);
  if(Number.isNaN(evert) || every <= 1){
    return domain;
  }
  return domain.filter((_, index) => index % every === 0);
}
```